pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1433,7 +1433,17 @@ private ByteBuf processMessageChunk(ByteBuf compressedPayload, MessageMetadata m
 
         ChunkedMessageCtx chunkedMsgCtx = chunkedMessagesMap.get(msgMetadata.getUuid());
 
-        if (msgMetadata.getChunkId() == 0 && chunkedMsgCtx == null) {
+        if (msgMetadata.getChunkId() == 0) {
```

Comment:
If the producer crashes and restarts, then send the previous message with `producer.newMessage.send()` API.
The message should be identified as a new message with a different uuid. See the solution at https://github.com/apache/pulsar/pull/20948.

Reply:
As we discussed internally, we don't need to use a different uuid here.